### PR TITLE
Fix race condition in reading the dropped spans number

### PR DESCRIPTION
**Pull request by bogdandrutu**
As any race condition this should be consider an undefined behavior, and a patch release should be done.

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: CHANGELOG.md
Comment (MrAlias) [inline CHANGELOG.md:13]:
```suggestion
- Fix race condition in reading the dropped spans number. (#2615)
```

File: sdk/trace/batch_span_processor.go
Comment (MrAlias) [inline sdk/trace/batch_span_processor.go:253]: It looks like `bsp.dropped` is never reset. Think there is a need to refactor the synchronization logic to ensure an atomic view into the whole batch being exported and then uniform reset. Until then, do we want to just remove logging this to fix the race condition?

Comment (bogdandrutu) [reply sdk/trace/batch_span_processor.go:253]: Not sure, my fix does not change any logic, it simply fixes the race. If you want more fixes, I don't have time to do, so up to @open-telemetry/go-maintainers to decide what to do.

**Review by pellared — APPROVED** [review]
While the synchornization logic behind  `bsp.dropped` could be improved I still see value in this PR.

**Review by XSAM — APPROVED** [review]
`atomic.LoadUint32(&bsp.dropped)` could be a marker that indicates a logging point for dropped spans while refactoring the synchronization logic.

**Review by MadVikingGod — APPROVED** [review]

File: sdk/trace/batch_span_processor.go
Comment (MadVikingGod) [reply sdk/trace/batch_span_processor.go:253]: @MrAlias When would it reset?  This is just a count of how many spans have been dropped, which is independent of the success or failure of subsequent batch exports.

If we did reset on export we wouldn't be getting a useful number because drops happen when the queue is full, which is most likely when the batch is locked during export.

**Comment by bogdandrutu** [comment]
Since this is an important bug fix, can you please merge and release 1.4.1. This blocks downstream dependencies to upgrade, since tools will block the upgrade because of race condition detected.

**Comment by MrAlias** [comment]
Working on the release now.

**Pull request merged**
